### PR TITLE
Stats highlight cards - add update timeframe notice

### PR DESCRIPTION
**Pull request by Addison-Stavlo**
Related to # Automattic/dotcom-forge#8781

## Proposed Changes

HOLD ON MERGING - for reduced cache for the daily highlights in D159517-code. Before this the new message is only true on annual stats and not daily.

* Adds an "Updates every 30 minutes" note to stats highlights card sliders headers.
* Adds a common component to be used as a shared wrapper for these headers, applies them to the weekly/monthly highlights cards and the annual highlights cards. The components used the same HTML elements and class names, but did not share/consolidate those via react components. 

BEFORE
Screenshot 2024-08-26 at 1 21 40 PM
Screenshot 2024-08-26 at 1 21 52 PM

AFTER

Screenshot 2024-08-26 at 1 22 01 PM
Screenshot 2024-08-26 at 1 22 11 PM

## Why are these changes being made?

* It feels broken if you look at these stats numbers but know you have a different amount of posts, comments, views, etc. No matter how much we reduce cache timers or bust cache, there are always race conditions and other items that could make these areas present slightly out of date information.
* Adding a small note about an update timer helps fight a feeling of brokenness, and makes stale values feel more acceptable.
* More context at pe7F0s-25n-p2

## Testing Instructions

* Go to stats in calypso and look at the traffic and insights tabs.
* Verify the cards slider headers have a sub header section mentioning "Updates every 30 minutes"

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: packages/components/src/highlight-cards/highlight-cards-heading.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { useTranslate } from 'i18n-calypso';
+
+export default function HighlightCardsHeading( { children }: { children: React.ReactNode } ) {
+	const translate = useTranslate();
+	return (
+		<div className="highlight-cards-heading__wrapper">
+			<h3 className="highlight-cards-heading">{ children }</h3>
+			<div className="highlight-cards-heaing__update-frequency">
```
Comment (allilevine) [inline packages/components/src/highlight-cards/highlight-cards-heading.tsx]: Is the class name missing a 'd' in heading? To be: `highlight-cards-heading__update-frequency`

Comment (Addison-Stavlo) [reply packages/components/src/highlight-cards/highlight-cards-heading.tsx]: Lol wow! Its crazy what the eyes overlook until pointed out. I carried that over to two different style rules as well. 🤣  Updated.

Comment (allilevine) [reply packages/components/src/highlight-cards/highlight-cards-heading.tsx]: Same 😅 At least you were consistent, so it still worked. 😂

**Review by allilevine — APPROVED** [review]
I'm seeing the "Updates every 30 minutes" info on Traffic and Insights:

Screen Shot 2024-09-04 at 4 57 35 PM

This will probably show up on Odyssey Stats too, but that's the correct behavior since they use the same endpoints and components.

**Comment by Addison-Stavlo** [comment]
I pushed a quick update for a translation check, so we can merge this without holding on translations.

If you swap to another locale, you should not see the blurb currently due to it not being translated:

Screenshot 2024-09-05 at 10 41 17 AM

**Comment by allilevine** [comment]
> If you swap to another locale, you should not see the blurb currently due to it not being translated:

Tested and I don't see the blurb in another locale:

Screen Shot 2024-09-05 at 10 56 38 AM

I still see the blurb in English:

Screen Shot 2024-09-05 at 10 57 09 AM

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16647035

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @Addison-Stavlo for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16647035).

**Pull request merged**